### PR TITLE
feat: add `UserSettings` component to design system

### LINKED ISSUE #783: Feature: Create `UserSettings` page component
### Type of feature

🍕 Feature

### Current behavior

image

image

### Suggested solution

This should start on the design system first

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by OgDev-01**
## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description
Th PR:

- creates a `LanguagePill` component to the atoms directory
- adds `LanguagePill` to design system
- adds the user settings screen to the storybook(design system)
- updates existing checkbox style to match the brand color

### Notes:

- This has no functionality yet, just the UI
- The current UI checkbox was replaced with the default checkbox in the design system

## Related Tickets & Documents
Fixes #783 

## Mobile & Desktop Screenshots/Recordings
image

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [x] 📕 storybook
- [ ] 🙅 no documentation needed

### CODE REVIEW

**Comment by bdougie** [comment]
I want to point out the importance of https://github.com/open-sauced/insights/issues/791 😂

This Page is not in the Page folder. Not a big deal, but just pointing out the confusion of our current design system setup.

Screen Shot 2023-01-19 at 6 34 19 AM

**Review by bdougie — COMMENTED** [review]
We should provide the list of timezones in the dropdown. 

This should be in a lib - https://stackoverflow.com/questions/38399465/how-to-get-list-of-all-timezones-in-javascript

Screen Shot 2023-01-19 at 6 40 51 AM

Perhaps it can be done in a new PR though

File: components/atoms/LanguagePill/LanguagePill.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import MLIcon from "/img/icons/🧠.svg";
+import RustIcon from "/img/icons/🦀.svg";
+
+interface LanguagePillProps {
+  topic: "react" | "javascript" | "python" | "ML" | "AI" | "rust" | string;
+  classNames?: string;
+  onClick?: () => void;
+}
+const LanguagePill = ({ topic, classNames, onClick }: LanguagePillProps) => {
+  const renderTopicIcon = (name: string) => {
```
Comment (brandonroberts) [inline components/atoms/LanguagePill/LanguagePill.tsx:17]: Let's make this a map instead so we don't have this growing list of nested ternaries.

```ts
const iconMap = {
  react: ReactIcon,
  // others
}

return iconMap[name] || ""
```

File: components/organisms/UserSettingsPage/user-settings-page.tsx [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+  return (
+    <div>
+      <div className="flex flex-col gap-4 text-sm text-light-slate-11">
+        <div>
+          <Title className="!text-2xl !text-light-slate-11" level={2}>
+            Public profile
+          </Title>
+          <form className="flex flex-col gap-6 mt-6">
+            <TextInput
+              classNames="bg-light-slate-4 text-light-slate-11 font-medium"
```
Comment (brandonroberts) [inline components/organisms/UserSettingsPage/user-settings-page.tsx:32]: Question: I thought we weren't using `font-medium` anymore?

```suggestion
              classNames="bg-light-slate-4 text-light-slate-11 font-medium"
```

File: components/atoms/LanguagePill/LanguagePill.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
+import React from "react";
+import Image from "next/legacy/image";
```
Comment (brandonroberts) [inline components/atoms/LanguagePill/LanguagePill.tsx:2]: Use non-legacy image

```suggestion
import Image from "next/image";
```

File: components/organisms/UserSettingsPage/user-settings-page.tsx [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+            />
+            <TextInput
+              classNames="bg-light-slate-4 text-light-slate-11 font-medium"
+              placeholder="april@stockgen.com"
+              label="Email*"
+            />
+            <div className="flex flex-col gap-3 ">
+              <label className="text-light-slate-11  font-normal">Email Preferences</label>
+              <Checkbox value={"true"} title="profile email" label="Display Email On Profile" />
+              <Checkbox value={"true"} title="profile email" label="Receive collaboration requests" />
```
Comment (brandonroberts) [inline components/organisms/UserSettingsPage/user-settings-page.tsx:44]: nit: update checkbox title

```suggestion
              <Checkbox value={"true"} title="collaboration requests" label="Receive collaboration requests" />
```

**Review by brandonroberts — CHANGES_REQUESTED** [review]

File: components/organisms/UserSettingsPage/user-settings-page.tsx (hunk 2 above)
Comment (OgDev-01) [reply components/organisms/UserSettingsPage/user-settings-page.tsx:32]: Ooops, my mistake 😇... I'm fixing now

**Comment by OgDev-01** [comment]
> We should provide the list of timezones in the dropdown.
> 
> This should be in a lib - https://stackoverflow.com/questions/38399465/how-to-get-list-of-all-timezones-in-javascript
> Perhaps it can be done in a new PR though

Opening a new issue to implement the form functionality, and this can be included also🍕

**Comment by brandonroberts** [comment]
@bdougie A general observation, but can we separate updating `Email Preferences` from  `Profile`. There isn't a clear distinction between the two sections.

**Comment by bdougie** [comment]
Yes, my original thought was adding it on the right. But wasn't sure if it would get buried on mobile.

**Comment by OgDev-01** [comment]
> Everything should float left on large screens.
> 
> Screen Shot 2023-01-19 at 12 31 40 PM

This is just the form component without a wrapper... once it is plugged into a page the wrapper will be used to handle the float 🍕

File: components/organisms/UserSettingsPage/user-settings-page.tsx [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+                <LanguagePill
+                  onClick={() => handleSelectInterest(topic)}
+                  classNames={`${selectedInterest.includes(topic) && "bg-light-orange-10 text-white"}`}
+                  topic={topic}
+                  key={index}
+                />
+              ))}
+            </div>
+            <button className="px-4 w-max py-2  rounded-lg bg-light-slate-4 border border-light-slate-8">
+              Update Interest
```
Comment (brandonroberts) [inline components/organisms/UserSettingsPage/user-settings-page.tsx:102]:
```suggestion
              Update Interests
```

**Review by brandonroberts — APPROVED** [review]

**Comment by brandonroberts** [comment]
We will adjust the timezone fields in implementation

**Pull request merged**